src/index.ts

```diff
@@ -1,3 +1,3 @@
 export * from './vanilla'
 export * from './react'
-export { default } from './react'
+export { default, createWithState } from './react'
```

Comment:
what's this `createWithState`?

Reply:
If you want to explicitly type the state, you also will have to explicitly type the stores too. But with `createWithState` you can explicitly type the state without having to explicitly type the stores.

```ts
// https://tsplay.dev/NlxqlN

// doesn't compile
create<{ count: number }>(
  withA("a",
    withB("b",
      storeIn => ({ count: 0 })
    )
  )
)

// compiles
create<{ count: number }, [[$$withA, string], [$$withB, string]]>(
  withA("a",
    withB("b",
      storeIn => ({ count: 0 })
    )
  )
)

// compiles
createWithState<{ count: number }>()(
  withA("a",
    withB("b",
      storeIn => ({ count: 0 })
    )
  )
)
```

This would be the recommended way to create stores users. I did mention it in #710 in the others section but didn't explain it.